### ISSUE TITLE
"Show Leo in the context menu on websites" setting does nothing

### ISSUE THREAD

**Issue by Infinoid** [brave/brave-browser#40629]
### Description

I enable this setting.  Then I right-click a website and it's not there.

### Steps to reproduce

1. Go to brave://settings/leo-assistant
2. Enable the "Show Leo in the context menu on websites" setting
3. Open a website
4. Right click it
5. No leo option in popup menu

### Actual result

![image](https://github.com/user-attachments/assets/8e4fcb8a-9a0f-4384-8998-c4a3a5ea30a1)

### Expected result

I was expecting to see a Leo in the context menu.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

Brave	1.69.153 Chromium: 128.0.6613.85 (Official Build) (64-bit) 
Revision	a69e96756922f88cbce3482971d70cfd1ee6f872
OS	Linux

### Channel information

- [X] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [ ] with Brave Shields disabled
- [ ] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by rebron** [comment]
@Infinoid When you select text, do you see the Leo AI tools in the context menu?

**Comment by Infinoid** [comment]
Yes, I do.  Have I misunderstood how the feature works, then?

**Comment by Infinoid** [comment]
When I use this text-selection context menu, it ignored my default model setting and used Mixtral again (see #40627).  But otherwise, it did the thing I asked it to.

**Comment by Infinoid** [comment]
Should "Leo AI tools" show up in the right-click menu when no page text is highlighted?  Maybe that's a feature request.

I had assumed it would be there, and the entire page text would be visible to the AI, but that doesn't seem to be how it works.  Apart from that, I don't think there's any bug to fix here.

**Comment by mhitza** [comment]
I agree it is confusing, just set up Brave with my self-hosted LLM and finally ready to give Leo a real shot.

My opinion is that actions such as Summarize and Explain should always be available in the contextual menu without selecting anything.

For the other options it makes sense to only show up when selecting text.

**Comment by ghost** [comment]
Found this while searching as to why Leo doesn't show up in right-click (context) menu. Very confusing indeed. If I'm reading an article and want to send it to my local (LAN) LLM to summarize, what's the proper workflow? Also when I right click on a link to an article I feel like Leo should show up there too so I can send that link to my local LLM while I continue reading other stuff.

**Comment by mattmcalister** [comment]
assigned to you @simonhong but feel free to reassign if you need to. the request is to show the AI options in the context menu list regardless of whether or not text has been selected on the page. if no text is selected then the whole page is the context.

**Comment by simonhong** [comment]
I think I'm not the right person for this issue. Maybe @fallaciousreasoning or @petemill ?
